services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/kafka/KafkaMessageCollector.java

```diff
@@ -92,26 +69,25 @@ public KafkaMessageCollector() {
         dependencies.add(IOFSwitchService.class);
         dependencies.add(IThreadPoolService.class);
         dependencies.add(KafkaMessageProducer.class);
+        dependencies.add(CommandProcessorService.class);
+        dependencies.add(InputService.class);
 
         return dependencies;
     }
 
     @Override
     public void init(FloodlightModuleContext moduleContext) {
-        configService.init(ConfigurationProvider.of(moduleContext, this));
-        commandContextFactory.init(moduleContext);
-        inputService.init(moduleContext);
+        // there is nothing to initialize here
     }
 
     @Override
     public void startUp(FloodlightModuleContext moduleContext) throws FloodlightModuleException {
         logger.info("Starting {}", this.getClass().getCanonicalName());
 
-        commandProcessor.init(moduleContext);
-        pingService.init(moduleContext);
+        ConfigurationProvider provider = ConfigurationProvider.of(moduleContext, this);
+        KafkaConsumerConfig consumerConfig = provider.getConfiguration(KafkaConsumerConfig.class);
 
-        KafkaConsumerConfig consumerConfig = configService.getConsumerConfig();
-        KafkaTopicsConfig topicsConfig = configService.getTopics();
+        KafkaTopicsConfig topicsConfig = moduleContext.getServiceImpl(KafkaMessageProducer.class).getTopics();
```

Comment:
Can we get rid of dependency between KafkaMessageCollector and KafkaMessageProducer? Please correct me if I'm wrong, but as I see we might get list of topics from KafkaConsumerConfig.

Reply:
This is second step in FL reorganisation. I am going to make one module that contain kafka producer service and service - factory for kafka consumers.

FL propose to split code across services (implement `IFloodlightService`), also it provide API to lookup any loaded service. So service is the only correct way to provide implement some functions. Services can handle requests produced by FL or by other services.

So if some service need to lookup topic name it should lookup and call some other service that provide this info. That why TopicsConfig are part of KafkaMessageProvider(some service).

PS Even if we ignore kafka-topics lookup, we can't drop KafkaMessageProducer(service) from here because KafkaMessageConsumer use it to send responses in kafka.